packages/channel/src/channel.ts

```diff
@@ -0,0 +1,34 @@
+import { Operation, spawn } from 'effection';
+import { Subscribable, Subscription, SymbolSubscribable, createSubscription, forEach } from '@effection/subscription';
+import { on, once } from '@effection/events';
+import { EventEmitter } from 'events';
+
+export class Channel<T> implements Subscribable<T, void> {
+  private bus = new EventEmitter();
+
+  [SymbolSubscribable](): Operation<Subscription<T, void>> {
+    return this.subscribe();
+  }
+
+  setMaxListeners(value: number) {
+    this.bus.setMaxListeners(value);
+  }
+
+  send(message: T) {
+    this.bus.emit('message', message);
+  }
+
+  subscribe(): Operation<Subscription<T, void>> {
+    let { bus } = this;
+    return createSubscription(function*(publish) {
+      yield spawn(forEach(on(bus, 'message'), function*([message]: [T]) {
+        publish(message)
+      }));
+      yield once(bus, 'close');
+    });
+  }
+
+  close() {
+    this.bus.emit('close');
```

Comment:
@jnicklas Something about this feels right in that closing a channel, completes all subscriptions. So we don't have to change anything about the `close()` method to make it a no-op, it just means that on subsequent calls, nobody is listening 😎 

What do you think about having `createChannel` be an operation that produces a channel resource. It's hard to imagine a case where you wouldn't want your channel to be a resource that shut down when it passes out of scope. It certainly would not be good to have subscriptions waiting for channels, so ensuring `close()` is always called seems like the right thing whether you're doing it manually or in a resource teardown.

Reply:
I'm a bit torn on this. I think it'll be rather common to have channels which never close (e.g. Atom in bigtest) and in those cases, having to allocate a channel via a resource is a bit inconvenient. Having *both* as an API might be a bit confusing.

Reply:
Given that we now have the ability to `close()` with an argument which is a win, I think that answers the question of always creating a channel as a resource, since an automatically closed channel can not know ahead of time what the argument to `close` should be.

In the event that you want to autoclose, it's up to the programmer about what to send to the `close()` method and they can manually do that when they create the channel as a resource..